packages/main/src/Menu.ts

```diff
@@ -375,8 +381,29 @@ class Menu extends UI5Element {
 			parentElement.selected = false;
 			(parentElement._popover.opener as HTMLElement)?.focus();
 		}
+		if (!menuItem.hasAttribute("ui5-menu-item")) {
+			return;
+		}
+		if (isUp(e)) {
+			this._handleNextOrPreviousItem(menuItem);
+		} else if (isDown(e)) {
+			this._handleNextOrPreviousItem(menuItem, true);
+		} else if (isLeft(e) || isRight(e)) {
+			menuItem._itemKeyDown(e);
+		}
 	}
 
+	_handleNextOrPreviousItem(menuItem: MenuItem, isNext?: boolean) {
+		const currentIndex = this._menuItems.indexOf(menuItem);
+		const nextItem = isNext ? this._menuItems[currentIndex + 1] : this._menuItems[currentIndex - 1];
+
+		if (nextItem) {
```

Comment:
You could transform this if else structure with a ternary operator into a one liner, in order to find the proper element to call `focus` upon.